go/integration_test/processor/create_test.go

```diff
@@ -451,6 +452,87 @@ func TestProcessor_CorrectAddressIsCreated(t *testing.T) {
 	}
 }
 
+func TestProcessor_CreateExistingAccountFails(t *testing.T) {
+	storageWithEntry := Storage{}
+	storageWithEntry[tosca.Key{42}] = tosca.Word{42}
+
+	tests := map[string]struct {
+		account Account
+		success bool
+	}{
+		"empty": {
+			account: Account{},
+			success: true,
+		},
+		"withCode": {
+			account: Account{Code: []byte{byte(vm.PUSH1), byte(0), byte(vm.RETURN)}},
+			success: false,
+		},
+		"withNonce": {
+			account: Account{Nonce: 1},
+			success: false,
+		},
+
+		// It is possible to create an account if it already had balance
+		"withBalance": {
+			account: Account{Balance: tosca.NewValue(42)},
+			success: true,
+		},
+
+		// Different to ethereum, on Sonic accounts can only have set storage if their nonce != 0
+		"withStorage": {
+			account: Account{Nonce: 1, Storage: storageWithEntry},
+			success: false,
+		},
+	}
+
+	for processorName, processor := range getProcessors() {
+		if strings.Contains(processorName, "floria") {
+			continue
+		}
+		for testName, test := range tests {
+			t.Run(processorName+"/"+testName, func(t *testing.T) {
+				checkValue := byte(42)
+				sender := tosca.Address{1}
+				addressToBeCreated := tosca.Address(crypto.CreateAddress(common.Address(sender), 0))
+
+				initCode := []byte{
+					byte(vm.PUSH1), checkValue,
+					byte(vm.PUSH1), byte(0),
+					byte(vm.MSTORE),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}
+				state := WorldState{
+					sender:             Account{},
+					addressToBeCreated: test.account,
+				}
+				transaction := tosca.Transaction{
+					Sender:    sender,
+					Recipient: nil,
+					GasLimit:  sufficientGas,
+					Input:     initCode,
+				}
+
+				transactionContext := newScenarioContext(state)
+
+				// Run the processor
+				result, err := processor.Run(tosca.BlockParameters{}, transaction, transactionContext)
+				if err != nil {
+					t.Errorf("execution failed with error %v", err)
+				}
+				if result.Success != test.success {
+					t.Errorf("execution success was %v, expected %v", result.Success, test.success)
+				}
+				if result.Success && !slices.Equal(result.Output, append(bytes.Repeat([]byte{0}, 31), checkValue)) {
+					t.Errorf("creation returned successful but with wrong output %v", result.Output)
+				}
+			})
```

Comment:
Unless this is already tested somewhere else, the following post-condition checks should be added:
- in case the account was created, the new account has a nonce of 1, no balance, empty storage, and the given code
- in case the account was not created, it should be checked that the resulting world state is the same as the initial state (except for the nonce of the account paying for the creation being increased by 1)